### PR TITLE
Sync besluiten

### PR DESCRIPTION
**Pull request by bbrands02**

### CODE REVIEW

File: Installation/installation.json [hunk 1]
```diff
@@ -83,15 +83,6 @@
...
             }
-          },
-          {
-            "https://vng.opencatalogi.nl/EntityEndpoint/ZaakInformatieObject.endpoint.json": {
-              "out": {
-                "body": {
-                  "mapping": "https://development.zaaksysteem.nl/mapping/xxllnc.XxllncZaakInformatieobjectResultOutMapping.mapping.json"
-                }
-              }
-            }
```
Comment (rjzondervan) [inline Installation/installation.json:94]: Could you reintroduce these lines?

File: src/Service/DocumentService.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+     *
+     * @return string|false $documentNumber Document number else false.
+     */
+    private function reserveDocumentNumber(Source $xxllncApi)
+    {
+        // Send the POST request to xxllnc.
+        try {
+            $response = $this->callService->call($xxllncApi, '/document/reserve_number', 'POST');
+            $result   = $this->callService->decodeResponse($xxllncApi, $response);
+            var_dump('Document number: ', $result['result']['instance']['serial']);
```
Comment (rjzondervan) [inline src/Service/DocumentService.php:130]: Var_dump

File: src/Service/ZGWToXxllncService.php [hunk 3]
```diff
@@ -490,108 +300,91 @@ public function mapZGWToXxllnc(string $casetypeId, ObjectEntity $zaakTypeObject,
         // Unset empty keys.
         $caseArray = array_filter($caseArray);
 
-        $sourceId = $this->sendCaseToXxllnc($caseArray, $synchronization);
-        if (!$sourceId) {
+        $sourceId = $this->sendCaseToXxllnc($caseArray, $caseObject, $synchronization);
+        if ($sourceId === false) {
+            var_dump("POST to xxllnc failed.");
```
Comment (rjzondervan) [inline src/Service/ZGWToXxllncService.php:305]: Shouldn't you put this in a log?

File: Installation/installation.json (hunk 1 above)
Comment (bbrands02) [reply Installation/installation.json:94]: This mapping does not exist in the bundle..

File: src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php [hunk 4]
```diff
@@ -1,14 +1,4 @@
...
-/**
- * This class handles the synchronization of one or more of xxllnc casetypes to zgw ztc zaaktypen.
- *
- * This ActionHandler executes the zaakTypeService->zaakTypeHandler.
- *
- * @author  Conduction BV <info@conduction.nl>, Barry Brands <barry@conduction.nl>
- * @license EUPL-1.2 https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
- *
- * @category ActionHandler
- */
```
[the remark this answers is not in the document]
Comment (bbrands02) [reply src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php:11]: They have been moved not deleted

File: src/Service/DocumentService.php (hunk 2 above)
Comment (bbrands02) [reply src/Service/DocumentService.php:130]: yes this work is in progress and shouldn't be triggered

File: src/Service/DocumentService.php [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        $multipart = [
+            [
+                'name'     => 'upload',
+                'contents' => $file_stream,
+                'filename' => $infoObject['informatieobject']['bestandsnaam'],
+            ],
+        ];
+
+        // Send the POST request to xxllnc.
+        var_dump('Request prepare file:');
```
[the remark this answers is not in the document]
Comment (bbrands02) [reply src/Service/DocumentService.php:95]: yes this work is in progress and shouldn't be triggered

File: src/Service/ZGWToXxllncService.php (hunk 3 above)
Comment (bbrands02) [reply src/Service/ZGWToXxllncService.php]: vardump removed, sendCaseToXxllnc should throw error logs

File: Installation/installation.json (hunk 1 above)
Comment (rjzondervan) [reply Installation/installation.json:94]: Ah, I forgot to rename it earlier, here's the correct $id:

`"$id": "https://development.zaaksysteem.nl/mapping/xxllnc.XxllncZaakInformatieobjectBodyOutMapping.mapping.json",`

File: src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (rjzondervan) [reply src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php:11]: I know, but wasn't the codacy convention to have these lines immediately after `<?php` ?

File: src/Service/DocumentService.php (hunk 2 above)
Comment (rjzondervan) [reply src/Service/DocumentService.php:130]: Alright

File: src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (bbrands02) [reply src/ActionHandler/ConnectBesluittypeToZaaktypeHandler.php:11]: Then I was confused probably

File: src/Service/BesluitToXxllncService.php [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+        $bsn             = ($zaakArrayObject['rollen'][0]['betrokkeneIdentificatie']['inpBsn'] ?? $zaakArrayObject['verantwoordelijkeOrganisatie'] ) ?? null;
+        if ($bsn === null) {
+            $this->logger->error('No bsn found in a rol->betrokkeneIdentificatie->inpBsn or verantwoordelijkeOrganisatie.');
+
+            return null;
+        }
+
+        $besluitObjectArray  = array_merge($besluitObject->toArray(), ['bsn' => $bsn, 'caseTypeId' => $besluittypeSourceId]);
+        $besluitMappingArray = $this->mappingService->mapping($xxllncZaakMapping, $besluitObjectArray);
+        unset($besluitMappingArray['zgwZaak']);
```
Comment (smisidjan) [inline src/Service/BesluitToXxllncService.php:177]: Maybe you can unset the zgwZaak in the mapping

File: src/Service/BesluitToXxllncService.php [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+     */
+    public function besluitToXxllncHandler(?array $data = [], ?array $configuration = []): array
+    {
+        $this->data          = $data;
+        $this->configuration = $configuration;
+
+        if (isset($this->data['response']['_self']['id']) === false || isset($this->data['response']['besluit']) === false) {
+            $this->logger->error('syncBesluitToXxllnc returned, no id or besluit found in given zaakbesluit.');
+            var_dump('died');
+            die;
```
Comment (smisidjan) [inline src/Service/BesluitToXxllncService.php:390]: can you remove this var_dump and die?

File: src/Service/ZGWToXxllncService.php [hunk 8]
```diff
@@ -88,297 +89,138 @@ class ZGWToXxllncService
...
+    // 'pip_authorized'         => true,
+    // 'send_auth_notification' => false,
+    // ];
+    // }
+    // }
+    // }
+    // }
+    // }//end if
+    // return $xxllncZaakArray;
+    // }//end mapPostRollen()
```
Comment (smisidjan) [inline src/Service/ZGWToXxllncService.php]: Can this function be deleted instead?

File: src/Service/ZGWToXxllncService.php [hunk 9]
```diff
@@ -396,90 +238,58 @@ private function getCaseObject(array $zaakArrayObject)
...
+        // If the zaakinformatieobjecten are already synced we need their source id (number) so we can add them to the case mapping.
+        // foreach ($zaakArrayObject['zaakinformatieobjecten'] as $key => $zaakInfoObject) {
+        // $zaakArrayObject['zaakinformatieobjecten'][$key]['xxllncDocumentNumber'] = $this->documentService->checkCustomNumber($zaakInfoObject, $this->xxllncAPI, 'zaakInfoObject');
+        // $zaakArrayObject['zaakinformatieobjecten'][$key]['xxllncReferenceId']    = $this->documentService->checkCustomNumber($zaakInfoObject, $this->xxllncAPI, 'enkelvoudigInfoObject');
+        // if ($zaakArrayObject['zaakinformatieobjecten'][$key]['xxllncDocumentNumber'] == null || $zaakArrayObject['zaakinformatieobjecten'][$key]['xxllncReferenceId'] == null) {
+        // unset($zaakArrayObject['zaakinformatieobjecten'][$key]);
+        // $this->logger->error("Ignoring infoobject {$zaakInfoObject['_self']['id']} because the document number or reference could not be created at the xxllnc api.");
+        // }
+        // break;
+        // }
```
Comment (smisidjan) [inline src/Service/ZGWToXxllncService.php:248]: Can this code be deleted instead?

**Review by smisidjan — CHANGES_REQUESTED** [review]

File: src/Service/ZGWToXxllncService.php (hunk 9 above)
Comment (bbrands02) [reply src/Service/ZGWToXxllncService.php:248]: This Id rather keep (in progress)

File: src/Service/BesluitToXxllncService.php (hunk 6 above)
Comment (bbrands02) [reply src/Service/BesluitToXxllncService.php]: Removed because not needed. This happens in a unset mapping

**Pull request merged**